### ISSUE TITLE
`lax.reduce` deviates from numpy's reduce on single-axis bitwise AND reductions

### ISSUE THREAD

**Issue by nicholasjng** [google/jax#11621]
This came up while working on #11115. In short, `lax.reduce` does not produce results consistent with numpy's `ufunc.reduce` facility when using single-axis reductions in combination with bitwise AND.

The following repro requires a checkout of #11115, since that has the code needed for `jax.lax.reduce` to allow integer operands in bitwise operations. (It is a little longer than it needs to be, but shows that the generated jaxprs are the same every time, and that bitwise OR/XOR work as expected in contrast to AND):

```python
import numpy as np
import jax
import jax.numpy as jnp
from jax import lax

lo, hi = np.iinfo(np.int32).min, np.iinfo(np.int32).max

key = jax.random.PRNGKey(0)
dtype = np.int32
and_initial = np.array(-1, dtype=dtype)
axis = (0,)
a = jax.random.randint(key, (3,4,5), lo, hi, dtype=jnp.int32)

np_res = np.bitwise_and.reduce(a, axis=axis, initial=and_initial, dtype=dtype)
jax_res = lax.reduce(a, and_initial, lax.bitwise_and, axis)

print(np_res)
print(jax_res)
fun = lambda operand: lax.reduce(operand, and_initial, lax.bitwise_and, axis)
print(jax.make_jaxpr(fun)(a))

axis2 = (0, 1)
np_res = np.bitwise_and.reduce(a, axis=axis2, initial=and_initial, dtype=dtype)
jax_res = lax.reduce(a, and_initial, lax.bitwise_and, axis2)

print(np_res)
print(jax_res)
fun = lambda operand: lax.reduce(operand, and_initial, lax.bitwise_and, axis2)
print(jax.make_jaxpr(fun)(a))


# bitwise or reduction test
or_initial = np.array(0, dtype=dtype)
np_res = np.bitwise_or.reduce(a, axis=axis, initial=or_initial, dtype=dtype)
jax_res = lax.reduce(a, or_initial, lax.bitwise_or, axis)

print(np_res)
print(jax_res)
fun = lambda operand: lax.reduce(operand, or_initial, lax.bitwise_or, axis)
print(jax.make_jaxpr(fun)(a))

# bitwise xor reduction test
np_res = np.bitwise_xor.reduce(a, axis=axis, initial=or_initial, dtype=dtype)
jax_res = lax.reduce(a, or_initial, lax.bitwise_xor, axis)

print(np_res)
print(jax_res)
fun = lambda operand: lax.reduce(operand, or_initial, lax.bitwise_xor, axis)
print(jax.make_jaxpr(fun)(a))
```

The part with `axis2`, which is a dual-axis reduction, was included to show that results do match when more than one axis is reduced. The same was observed during the PR review of #11115 - all tests pass for more than one reduction axis, but fail for all integer dtypes with a single axis.

**Comment by jakevdp** [comment]
It looks like your PR doesn't touch this line: https://github.com/google/jax/blob/main/jax/_src/lax/lax.py#L3717-L3718

Which I think means that the operation is using the incorrect identity when run under JIT.

**Comment by nicholasjng** [comment]
Wow, great catch once again. And that's a bingo, all tests pass. Updating the PR branch now.
